source/SoundBufferBase.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+class SoundBufferBase
+{
+public:
+	typedef SoundBufferBase* (*CreateSoundBufferFunc)(void);
+	static CreateSoundBufferFunc Create;
```

Comment:
This works, but I find there are already too many global variables and initialisation code and if one could instead do the same as the network stack, and add a virtual function to `FrameBase` as in https://github.com/audetto/AppleWin/blob/e0bb9f14c1ba8b6961a1b82c5e8981a6ade29871/source/FrameBase.h#L49